projects/wave-core-new/src/lib/layers/legend/legend-vector/vector-legend.component.ts

```diff
@@ -1,32 +1,104 @@
-import {ChangeDetectionStrategy, Component, Input} from '@angular/core';
-import {Symbology, SymbologyType} from '../../symbology/symbology.model';
-
+import {ChangeDetectionStrategy, Component, Input, OnInit} from '@angular/core';
+import {
+    ColorParam,
+    DerivedColor,
+    DerivedNumber,
+    LineSymbology,
+    NumberParam,
+    PointSymbology,
+    PolygonSymbology,
+    SymbologyType,
+    VectorSymbology,
+} from '../../symbology/symbology.model';
 import {PolygonIconStyle} from '../../layer-icons/polygon-icon/polygon-icon.component';
 import {PointIconStyle} from '../../layer-icons/point-icon/point-icon.component';
-import {LineIconStyle} from '../../layer-icons/line-icon/line-icon.component';
-
-interface IconValue {
-    icon: PointIconStyle | LineIconStyle | PolygonIconStyle;
-    value: string | number;
-}
+import {VectorLayer} from '../../layer.model';
+import {BLACK, Color, WHITE} from '../../../colors/color';
+import {ColorBreakpoint} from '../../../colors/color-breakpoint.model';
 
 @Component({
     selector: 'wave-vector-legend',
     templateUrl: 'vector-legend-component.html',
     styleUrls: ['vector-legend.component.scss'],
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class VectorLegendComponent {
+export class VectorLegendComponent implements OnInit {
     readonly ST = SymbologyType;
     @Input()
     showDefaultStyle = true;
 
-    @Input()
-    symbology!: Symbology;
-    fillColorAttribute: string | undefined;
-    fillStyles: Array<IconValue> = [];
-    strokeColorAttribute: string | undefined;
-    strokeStyles: Array<IconValue> = [];
+    @Input() layer!: VectorLayer;
+
+    symbology!: VectorSymbology;
```

Comment:
```suggestion
    symbology?: VectorSymbology;
```

`!` just tells TypeScript that there is a value, but you only set it as ngOnInit which means it is unset after construction until ngOnInit is called.

Reply:
changed that as well: 3d5f966e761ff9e0888ee8ff23d48968641d47e2